android/app/src/main/java/com/mattermost/rnbeta/NotificationReplyBroadcastReceiver.java

```diff
@@ -114,11 +117,15 @@ public void onResponse(Call call, final Response response) throws IOException {
     }
 
     protected String buildReplyPost(String channelId, String rootId, String message) {
-        return "{"
-                + "\"channel_id\": \"" + channelId + "\","
-                + "\"message\": \"" + message + "\","
-                + "\"root_id\": \"" + rootId + "\""
-                + "}";
+        try {
+            JSONObject json = new JSONObject();
+            json.put("channel_id", channelId);
+            json.put("message", message);
+            json.put("root_id", rootId);
+            return json.toString();
+        } catch(JSONException e) {
+            return "{}";
```

Comment:
I guess `JSONObject.put(String name, Object value)` only throws if `name` is `null` or `value` is something other than a string, which should never happen? Nevertheless, how does the server behave if it receives an empty request body; after the roundtrip, does this trigger an error message visible to the user?

Reply:
Server will respond with a 400 and user will be presented with a “Message faild to send”